### PR TITLE
Implemented the Search feature

### PR DESCRIPTION
**Pull request by clovellbsc**
## Before you start

1. Have you run `git pull origin main` on your branch to bring in the latest changes?
2. Is there an issue on the repository for this request? If not, please create one.
3. Have you followed the correct workflow in the [team documentation](https://docs.google.com/document/d/13yi709ax4nmfTG9J0F57yEFpYRH7fdtczwvVJCrNIsM/edit#)?
4. Conflict resolution to be dealt with **on GitHub**.

## What do these code changes do?
Created new full stack feature for searching for users
Created route, controller and updated the app.js
created a search page through handlebars

## How do these code changes affect the application?
search 
Ability to use search bar to find users by either first or last name
search results display users names and profile picture
posts display with the profile picture of the user who posted it

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [ ] I have created a PR ticket on the [Trello board](https://trello.com/b/P8lEsvZe/acebook).
- [ ] I will link the PR to the Trello card once created.
- [ ] I will send the PR in the Slack channel and ping @here.

### CODE REVIEW

**Comment by heykathl** [comment]
[I would like to be able to search for people I know.](https://trello.com/c/p98nbQcq/30-i-would-like-to-be-able-to-search-for-people-i-know)

File: app.js [hunk 1]
```diff
@@ -71,6 +72,7 @@ app.use("/", homeRouter);
 app.use("/posts", sessionChecker, postsRouter);
 app.use("/sessions", sessionsRouter);
 app.use("/users", usersRouter);
+app.use("/search", sessionChecker, searchRouter);
```
Comment (PKilgarriff) [inline app.js:75]: Good idea, nice use of the existing middleware

File: controllers/search.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    });
+  },
+  Create: (req, res) => {
+    const searchArray = req.body.message.replace(/[\W_]+/g, " ").split(/[ ,]+/);
+    const findQuery = [];
+    searchArray.forEach((name) => {
+      const first = { firstName: { $regex: name, $options: "i" } };
+      const second = { lastName: { $regex: name, $options: "i" } };
+      findQuery.push(first, second);
+    });
```
Comment (PKilgarriff) [inline controllers/search.js:16]: Saw this working, plus it should be expandable if we wanted to search other fields

File: cypress/integration/searching_users_spec.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
+describe("Searching users", () => {
+  beforeEach(() => {
+    cy.signUp();
+    cy.signUp("Chris", "Coding", "chris@coding.com", "12345");
+    cy.signUp("Paul", "Coding", "paul@coding.com", "12345");
```
Comment (PKilgarriff) [inline cypress/integration/searching_users_spec.js:5]: Can't believe you're committing my password in plain text, cheers Chris

File: views/posts/index.hbs [hunk 4]
```diff
@@ -16,17 +16,24 @@
...
 
-      <div class="post-header">
-        <span class="name">{{this.user.firstName}} {{this.user.lastName}}</span>
-        <span class="date-posted">15 mins ago</span>
-      </div>
+      <div class="profile-picture">
+          <img src={{this.user.profilePicture}} width="100"
+            height="100"
+            class="rounded-circle"/>
+        </div>
```
Comment (PKilgarriff) [inline views/posts/index.hbs:23]: This is great! Also think you've basically single-handedly done a whole user story there

**Review by PKilgarriff — APPROVED** [review]
Looks good, I might need to brush up on regex as well. Are the search queries case sensitive? Might be worth an edge case test, but in general it's great - cheers for doing what turned out to be such a mammoth feature!
